### ISSUE TITLE
Adapted Ascend npu for LongContextAttention\UlyssesAttention\RingAttention calulation based on torch_npu.npu_fusion_attention_v2 & npu_fusion_attention_grad_v2 for training and inference case

### ISSUE THREAD

**Issue by L4-1024** [feifeibear/long-context-attention#166]
base on the former issue: https://github.com/feifeibear/long-context-attention/pull/162

since the `torch_npu.npu_fused_infer_attention_score`  not suitable for the trainning case and backward calculation.  It can produce the following warning
`[rank0]:[W1207 02:51:33.024214569 VariableFallbackKernel.cpp:67] Warning: **npu::npu_fused_infer_attention_score: an autograd kernel was not registered to the Autograd key(s)** but we are trying to backprop through it. This may lead to silently incorrect behavior. This behavior is deprecated and will be removed in a future version of PyTorch. If your operator is differentiable, please ensure you have registered an autograd kernel to the correct Autograd key (e.g. DispatchKey::Autograd, DispatchKey::CompositeImplicitAutograd). If your operator is not differentiable, or to squash this warning and use the previous behavior, please register torch::CppFunction::makeFallthrough() to DispatchKey::Autograd. (function operator())`

So to optimize this case, and to unified the calculation of  `LongContextAttention\UlyssesAttention\RingAttention`  on the Ascend, we use the kernels of `torch_npu.npu_fusion_attention_v2 & npu_fusion_attention_grad_v2` for the forwad and the backward. 
the `torch_npu.npu_fusion_attention_v2`  implements the fused computation logic for Transformer attention, where computations such as matmul, transpose, add, softmax, dropout etc. are fused into a unified process.

A test script is included; run it with:
`sh scripts/run_hybrid_npu.sh`
`sh scripts/run_npu.sh`

it will shows the following result:

`out: max 0.1591796875, mean 0.021240234375

out diff:
[0] max 0.0, mean 0.0
[1] max 0.0, mean 0.0
[2] max 0.0, mean 0.0
[3] max 0.0, mean 0.0`
